# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/COLRFonts.h
# Commit: 02d6294b7a6d
# Full Hash: 02d6294b7a6d4e58e12ad1ca7259eb2cbdf53ce1
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2022-08-17 09:10:29
# Regressor Bug: 1740530
# File Overlap Count: 1
# Description:
#   Bug 1740530 - patch 4 - Use harfbuzz color-palette access functions, instead of rolling our own. r=gfx-reviewers,lsalzman
#   
#   Depends on D152038
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D153870
# ==============================================================================

diff -r 673f9010ea75 -r 02d6294b7a6d gfx/thebes/COLRFonts.h
--- a/gfx/thebes/COLRFonts.h	Tue Aug 16 12:48:32 2022 +0000
+++ b/gfx/thebes/COLRFonts.h	Tue Aug 16 12:48:33 2022 +0000
@@ -9,6 +9,7 @@
 #include "mozilla/gfx/2D.h"
 
 struct hb_blob_t;
+struct hb_face_t;
 
 namespace mozilla {
 
@@ -28,7 +29,7 @@
                                                    uint32_t aGlyphId);
 
   static bool PaintGlyphLayers(
-      hb_blob_t* aCOLR, hb_blob_t* aCPAL, const COLRBaseGlyphRecord* aLayers,
+      hb_blob_t* aCOLR, hb_face_t* aFace, const COLRBaseGlyphRecord* aLayers,
       DrawTarget* aDrawTarget, layout::TextDrawTarget* aTextDrawer,
       ScaledFont* aScaledFont, DrawOptions aDrawOptions,
       const sRGBColor& aCurrentColor, const Point& aPoint);